### DISCUSSION TITLE
The field array value on the input did not show up after some conditional rendering

### DISCUSSION THREAD

**Discussion by kaisiant** [react-hook-form/react-hook-form#3029] [Q&A] [answered]
Take a look at this CSB: https://codesandbox.io/s/react-hook-form-with-ui-library-forked-scg3i?file=/src/index.js

So here is the case, when I select  "enabled", the array field will be displayed and the user can type something on it.
But when I select "disabled" and then "enabled" again, the values on array field is gone. By right, the values should be there coz when I still able to see the values on console.

**Comment by bluebill1049** [comment, marked as answer]
where is the "enabled"?

**Reply by kaisiant** [reply]
sorry, the CSB just an example. 
"enabled" = when select iceCreamType (vanilla)
"disabled" = when select other iceCreamType

**Reply by bluebill1049** [reply]
duplicated: https://github.com/react-hook-form/react-hook-form/discussions/3023#discussioncomment-87392

**Reply by kaisiant** [reply]
> duplicated: [#3023 (reply in thread)](https://github.com/react-hook-form/react-hook-form/discussions/3023#discussioncomment-87392)

Not exactly duplicated.
His case was conditional rendering the fields which are not fieldArray and the problem he was facing is the wrong value persist on the input as the third input's value depends on the second input's value. His case can be easily solved by adding a key at controller.

In my case, I am conditional rendering normal field and fieldArray, the value didn't show up in the input when I toggled the status from "enabled" to "disabled" then to "enabled" again. But I solved my problem by referring to the conditional field array CSB which is using useWatch.

Hope you can take a look at the CSB, just in case I am using it in the wrong way.

**Reply by bluebill1049** [reply]
honestly, that was confusing... you forked someone's else CSB which was reported couple days, and then added your stuff on top of it, and yet there is no steps to reproduce. It doesn't make my life any easier.

for your issue perhaps try `shouldUnregister: false`

**Reply by kaisiant** [reply]
> honestly, that was confusing... you forked someone's else CSB which was reported couple days, and then added your stuff on top of it, and yet there is no steps to reproduce. It doesn't make my life any easier.
> 
> for your issue perhaps try `shouldUnregister: false`

Which CSB are you referring to, https://codesandbox.io/s/react-hook-form-with-ui-library-forked-scg3i?file=/src/index.js this one I forked from the official docs.

1. select ice cream type "vanilla", array field will be displayed
2. type "something" on it
3. select other ice cream type, then select back the "vanilla"
4. (expected) "something" is shown on the input by using **useWatch**.
     (before)     "something" is not shown on the input.

**Reply by bluebill1049** [reply]
nvm, i got confused with the other's CSB.

**Reply by bluebill1049** [reply]
<img width="803" alt="Screen Shot 2020-09-30 at 1 54 22 pm" src="https://user-images.githubusercontent.com/10513364/94641364-7285a800-0324-11eb-9339-69963f1a29ac.png">

if a component gets unmounted and its data gets lost as well by default at react hook form, unless you set to shouldUnregister: false at useFOrm.

**Reply by kaisiant** [reply]
> if a component gets unmounted and its data gets lost as well by default at react hook form, unless you set to shouldUnregister: false at useFOrm.

Suggested solution (**shouldUnregister: false**): https://codesandbox.io/s/react-hook-form-with-ui-library-forked-ci7hc?file=/src/index.js
(Expected) - value show up
(Actual) - value did not show up

Working solution (**useWatch**): https://codesandbox.io/s/react-hook-form-with-ui-library-forked-scg3i?file=/src/index.js

I  guess I just need to stick with **useWatch** will do

**Reply by bluebill1049** [reply]
👍
